tests/test_benchmark.py

```
@@ -0,0 +1,171 @@
+            db=f"sqlite:///{db_path}",
+            lock=Lock(str(lock_path)),
+            db_lock=Lock(str(db_lock_path)),
+            tqdm_kwargs={"disable": True},
+        )
+
+        list(worker)  # iterate
+
+    benchmark(iterate_and_measure)
+
+
+def test_dag_structure_validation(tmp_path):
```

Ruff (0.14.5)

Unused function argument: `tmp_path`

(ARG001)